### PR TITLE
[Core-973] Remove data stringification

### PR DESCRIPTION
**Pull request by hli-navarik**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -44,6 +47,7 @@ export class Storage<MetaType extends object> implements StorageInterface<MetaTy
     this.logger = logger || defaultLogger
     this.logger.debug({ component: "Storage" }, `Initializing storage (cache size: ${cacheSize}, static schemas: ${schema.length}, static data: ${data.length})`)
 
+    this.schemas = {}
```
Comment (nav-achernikov) [inline src/index.ts:50]: You should be able to get the schema from `this.ddl`

File: src/query-parser.ts [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
     }
 
     const args: Array<any> = []
     for (const field in searchParams) {
+      const typeLookUp = field =="type" ? null: schemas[field]
       const value = searchParams[field]
       const term = value.operator
-        ? value
-        : { operator: "literal", args: [value] }
+        ? castType(value, typeLookUp)
```
Comment (nav-achernikov) [inline src/query-parser.ts:27]: const value = castType(searchParams[field], typeLookUp)
const term = value.operator
        ? value
        : { operator: "literal", args: [value] }

File: src/query-parser.ts [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+  }
+  else if (type.includes('boolean')) {
+    return Boolean(value)
+  }
+  else return value
+}
+
 export class QueryParser {
-  parse(searchParams: Dictionary<any>): SearchQuery {
+  parse(searchParams: Dictionary<any>, schemas: object): SearchQuery {
```
Comment (nav-achernikov) [inline src/query-parser.ts:16]: Do `schemas` change with every call to `parse()`? if not, then they should be the QueryParser's constructor argument

File: src/query-parser.ts [hunk 4]
```diff
@@ -1,18 +1,31 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const castType = (value, type) => {
+  if (!type){return value}
+  if (type.includes('int') || type.includes('long')) {
+    return Number(value)
```
Comment (nav-achernikov) [inline src/query-parser.ts:7]: What about floats and doubles?

File: src/query-parser.ts [hunk 5]
```diff
@@ -1,18 +1,31 @@
 import { Dictionary } from "@navarik/types"
 import { SearchQuery } from "./types"
 
+const castType = (value, type) => {
+  if (!type){return value}
+  if (type.includes('int') || type.includes('long')) {
```
Comment (nav-achernikov) [inline src/query-parser.ts:6]: These `ifs` are a perfect case for the strategy pattern (see operator resolution in the query builder)

**Review by nav-achernikov — CHANGES_REQUESTED** [review]

File: src/query-parser.ts (hunk 2 above)
Comment (hli-navarik) [reply src/query-parser.ts:27]: Should have caught that. lazy brain

File: src/index.ts (hunk 1 above)
Comment (hli-navarik) [reply src/index.ts:50]: I am able to get each schema by type with `describe` one by one, but not altogether; hence, I figured I would work with the `schema` input instead.

Comment (nav-achernikov) [reply src/index.ts:50]: not all the schemas are provided statically to storage. In fact, Core provides none.
this.ddl is the schema registry you should be using

File: src/query-parser.ts [hunk 6]
```diff
@@ -1,15 +1,39 @@
...
+  float: (value) => Number(value),
+  long: (value) => Number(value),
+  boolean: (value) => Boolean(value)
+}
+
+const castType = (value, type) => {
+  if (!type) {
+    return value
+  }
+  const castFunction = dataTypes[type.split(' ')[0]]
```
Comment (nav-achernikov) [inline src/query-parser.ts:16]: oh... damn. this isn't good =) The fact that you need a split here is not a very good sign

File: src/query-parser.ts [hunk 7]
```diff
@@ -1,15 +1,39 @@
...
   parse(searchParams: Dictionary<any>): SearchQuery {
+
     if (searchParams.operator && searchParams.args) {
       return <SearchQuery>searchParams
     }
 
     const args: Array<any> = []
     for (const field in searchParams) {
-      const value = searchParams[field]
+      const typeLookUp = field == "type" ? null : this.schemas[field]
```
Comment (nav-achernikov) [inline src/query-parser.ts:35]: type is not the only field from CanonicalEntity that's not described by the schema

**Review by nav-achernikov — CHANGES_REQUESTED** [review]

File: src/query-parser.ts (hunk 6 above)
Comment (hli-navarik) [reply src/query-parser.ts:16]: it should be split(',').... it's to split something like 'string, required'

**Review by nav-achernikov — APPROVED** [review]

**Pull request merged**